pkg/minikube/cluster/cluster.go

```diff
@@ -222,10 +223,15 @@ func createDriverOptions(driver drivers.Driver, explicitOptions map[string]inter
 	return checkFlags, nil
 }
 
+// CacheMinikubeISOFromURL download minishift ISO from a given URI.
+// It also checks sha256sum if present and then put ISO to cached directory.
 func (m *MachineConfig) CacheMinikubeISOFromURL() error {
 	fmt.Println(fmt.Sprintf("Downloading ISO '%s'", m.MinikubeISO))
+	tmpISOFile, err := ioutil.TempFile(filepath.Join(constants.Minipath, "cache", "iso"), "minishift.part")
```

Comment:
Let's make this 

    tmpISOFile, err := os.Create(m.GetISOCacheFilepath() + ".part")

Then one has some understandable filenames while download is in process, but also when it fails.